internal/controller/server/execution/execution_test.go

```diff
@@ -76,4 +79,137 @@ func Test_execution_SaveMetadataToDb(t *testing.T) {
 			}
 		})
 	}
+}*/
+
+
+func TestExecuteJob(t *testing.T) {
+	jobExecutorRepoMock:=new(job.JobExecutorMock)
+	metadataRepoMock := new(metadata.MetadataMock)
+	mockScheduler := new(scheduler.SchedulerMock)
+	mockRandomIdGenerator := new(idgen.IdGeneratorMock)
+	executorRepoMock :=	 new(executor.ExecutorMock)
+
+	execution:=NewExec(metadataRepoMock,executorRepoMock,jobExecutorRepoMock,mockRandomIdGenerator,mockScheduler)
+
+	testJobData:= map[string]string {
+		"env1": "envValue1",
+		"env2": "envValue2",
+	}
+
+	mockScheduler.On("AddToPendingList", uint64(11)).Return(nil)
+	mockRandomIdGenerator.On("Generate").Return(uint64(11),nil)
+	jobExecutorRepoMock.On("ExecuteJob","11","testJob",testJobData).Return(nil)
+	jobExecutorRepoMock.On("CheckJobMetadataIsAvailable","testJob").Return(true,nil)
+
+	jobId,err:= execution.ExecuteJob(context.Background(),"testJob",testJobData)
+	assert.Nil(t,err)
+	assert.Equal(t,uint64(11),jobId)
+	jobExecutorRepoMock.AssertExpectations(t)
+	mockScheduler.AssertExpectations(t)
+	mockRandomIdGenerator.AssertExpectations(t)
 }
+
+func TestExecuteJobForJobExecutorRepoFailure(t *testing.T) {
+	jobExecutorRepoMock:=new(job.JobExecutorMock)
+	metadataRepoMock := new(metadata.MetadataMock)
+	mockScheduler := new(scheduler.SchedulerMock)
+	mockRandomIdGenerator := new(idgen.IdGeneratorMock)
+	executorRepoMock :=	 new(executor.ExecutorMock)
+
+	execution:=NewExec(metadataRepoMock,executorRepoMock,jobExecutorRepoMock,mockRandomIdGenerator,mockScheduler)
+
+	testJobData:= map[string]string {
+		"env1": "envValue1",
+		"env2": "envValue2",
+	}
+	mockScheduler.On("AddToPendingList", uint64(11)).Return(nil)
+	mockRandomIdGenerator.On("Generate").Return(uint64(11),nil)
+	jobExecutorRepoMock.On("ExecuteJob","11","testJob",testJobData).Return(errors.New("failed to execute job"))
+	jobExecutorRepoMock.On("CheckJobMetadataIsAvailable","testJob").Return(true,nil)
+
+	jobId,err:= execution.ExecuteJob(context.Background(),"testJob",testJobData)
+	assert.Equal(t,"failed to execute job",err.Error())
+	assert.Equal(t,uint64(11),jobId)
+	jobExecutorRepoMock.AssertExpectations(t)
+	mockScheduler.AssertExpectations(t)
+	mockRandomIdGenerator.AssertExpectations(t)
+}
+
+func TestExecuteJobForSchedulerFailure(t *testing.T) {
+	jobExecutorRepoMock:=new(job.JobExecutorMock)
```

Comment:
have you run go fmt ./... ? , Looks like space is inconsistent at a few places including line 139.